vector/src/main/java/im/vector/app/features/poll/create/CreatePollFragment.kt

```diff
@@ -0,0 +1,132 @@
+        return FragmentCreatePollBinding.inflate(inflater, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        vectorBaseActivity.setSupportActionBar(views.createPollToolbar)
+
+        views.createPollRecyclerView.configureWith(controller, disableItemAnimation = true)
+        controller.callback = this
+
+        views.createPollClose.debouncedClicks {
+            requireActivity().finish()
+        }
+
+        views.createPollButton.debouncedClicks {
+            viewModel.handle(CreatePollAction.OnCreatePoll)
+        }
+
+        viewModel.subscribe(this) {
+            views.createPollButton.isEnabled = it.canCreatePoll
+        }
+
+        viewModel.observeViewEvents {
+            when (it) {
+                CreatePollViewEvents.Success                  -> handleSuccess()
+                CreatePollViewEvents.EmptyQuestionError       -> handleEmptyQuestionError()
+                is CreatePollViewEvents.NotEnoughOptionsError -> handleNotEnoughOptionsError(it.requiredOptionsCount)
+            }
+        }
+    }
+
+    override fun invalidate() = withState(viewModel) {
+        controller.setData(it)
+    }
+
+    override fun onQuestionChanged(question: String) {
+        viewModel.handle(CreatePollAction.OnQuestionChanged(question))
+    }
+
+    override fun onOptionChanged(index: Int, option: String) {
+        viewModel.handle(CreatePollAction.OnOptionChanged(index, option))
+    }
+
+    override fun onDeleteOption(index: Int) {
+        viewModel.handle(CreatePollAction.OnDeleteOption(index))
+    }
+
+    override fun onAddOption() {
+        viewModel.handle(CreatePollAction.OnAddOption)
+        // Scroll to bottom to show "Add Option" button
+        views.createPollRecyclerView.apply {
+            postDelayed({
+                smoothScrollToPosition(adapter?.itemCount?.minus(1) ?: 0)
+            }, 100)
+        }
+    }
+
+    private fun handleSuccess() {
+        requireActivity().finish()
+    }
+
+    private fun handleEmptyQuestionError() {
+        renderToast(getString(R.string.create_poll_empty_question_error))
+    }
+
+    private fun handleNotEnoughOptionsError(requiredOptionsCount: Int) {
+        renderToast(
+                resources.getQuantityString(
+                        R.plurals.create_poll_not_enough_options_error,
+                        requiredOptionsCount,
+                        requiredOptionsCount
+                )
+        )
+    }
+
+    private fun renderToast(message: String) {
+        views.createPollToast.removeCallbacks(hideToastRunnable)
+        views.createPollToast.text = message
+        views.createPollToast.isVisible = true
+        views.createPollToast.postDelayed(hideToastRunnable, 2_000)
+    }
+
+    private val hideToastRunnable = Runnable {
+        views.createPollToast.isVisible = false
```

Comment:
I think this can crash if the Fragment has been destroyed

Reply:
We are using the same mechanism while recording voice messages. It is not crashing because it uses Handler inside. Also, we will never show toast on this screen, since we disable the button.

But I want to keep this to protect against post improvements.